WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -11,21 +11,36 @@
 
 namespace WalletWasabi.Services
 {
-	public class SystemAwakeChecker : PeriodicRunner
+	public class SystemAwakeChecker
```

Comment:
Yes, I believe we don't really need `PeriodicRunner`. 

However, please note that `IPowerSavingInhibitorTask.Prolong` must be removed for this PR to work. It is actually reasonable I think because we don't really need `Prolong` AFAIK.

Reply:
Also `SystemAwakeChecker` should be renamed to something like `SystemAwakeService` or something because if I understand your proposal correctly, it should just do what others tell it to do and not check anything itself.

Reply:
"`IPowerSavingInhibitorTask.Prolong` must be removed for this PR to work."
Can you elaborate on this? That would prevent the sleep/hibernation when CJ is in progress, but not in CriticalPhase.
Something is missing for that to work? I tried not to change your code too much.
 Your `ActionAsync` is basicly the `PreventSleepAsync()` now and I didn't change the inside of it. (Apart from one if statement)

Reply:
SideNote: `CoinJoinManager` won't know about the `SystemawakeChecker`, so this goes back to being a `HostedService`.